cli/tests/unit/test_config_resolver.py

```diff
@@ -0,0 +1,233 @@
+            ConfigLoader,
+            "_download_semgrep_cloud_platform_scan_config",
+            side_effect=[
+                SemgrepError(
+                    "Failed to download configuration. HTTP 500 when fetching URL"
+                ),
+            ],
+        )
+
+        patched_fallback_download = mocker.patch.object(
+            ConfigLoader,
+            "_download_semgrep_cloud_platform_fallback_scan_config",
+            return_value=config_file,
+        )
+
+        config = config_loader._fetch_semgrep_cloud_platform_scan_config()
+
+        assert config == config_file
+        assert patched_download.call_count == 1
+        assert patched_fallback_download.call_count == 1
+
+    @pytest.fixture
+    def mocked_state(self, mocker):
+        mocked = mocker.MagicMock()
+        mocked.request_id = uuid4()
+        mocked.env.semgrep_url = API_URL
+        mocker.patch("semgrep.config_resolver.get_state", return_value=mocked)
+        return mocked
+
+    @pytest.fixture
+    def mocked_cloud_platform_request(
+        self, mocked_state: SemgrepState, config_loader: ConfigLoader
+    ) -> out.ScanRequest:
+        products = [
+            out.Product.from_json(PRODUCT_NAMES[p])
+            for p in config_loader._config_path.split(",")
+        ]
+
+        request = out.ScanRequest(
+            meta=out.RawJson({}),
+            scan_metadata=out.ScanMetadata(
+                cli_version=out.Version(__VERSION__),
+                unique_id=out.Uuid(str(mocked_state.request_id)),
+                requested_products=products,
+                dry_run=True,
+            ),
+            project_metadata=config_loader._project_metadata_for_standalone_scan(),
+        )
+
+        return request
+
+    @pytest.fixture
+    def mocked_scan_response(self, config_loader) -> out.ScanResponse:
+        scan_info = out.ScanInfo(
+            enabled_products=[
+                out.Product.from_json(PRODUCT_NAMES[p])
+                for p in config_loader._config_path.split(",")
+            ],
+            deployment_id=1,
+            deployment_name="r2c",
+        )
+        scan_config = out.ScanConfiguration(rules=out.RawJson(value={}))
+        engine_config = out.EngineConfiguration()
+
+        return out.ScanResponse(
+            info=scan_info,
+            config=scan_config,
+            engine_params=engine_config,
+        )
+
+    @pytest.mark.quick
+    @pytest.mark.osemfail
+    def test__download_semgrep_cloud_platform_scan_config_success(
+        self,
+        config_loader: ConfigLoader,
+        mocked_state,
+        mocked_cloud_platform_request: out.ScanRequest,
+        mocked_scan_response: out.ScanResponse,
+        mocker,
+    ):
+        mock_response = mocker.MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = mocked_scan_response.to_json()
+        mocked_state.app_session.post.return_value = mock_response
```

Comment:
It's much better to use the `requests_mock` plugin/fixture to mock HTTP requests than stubbing request objects by hand. Theres loads of examples in `test_ci.py`

If you're open to it, I'd prefer that we take that approach here.

Reply:
Done in [fa09f29](https://github.com/semgrep/semgrep/pull/9205/commits/fa09f298b53944146f936b6f4eed6c69d2dab22a) and [9c920ba](https://github.com/semgrep/semgrep/pull/9205/commits/9c920bad1882e74f5d0e2f83a1d5bb94b75c0732)